### PR TITLE
feat(ui-number-input): add renderIcons prop

### PR DESCRIPTION
**Pull request by szalonna**
Closes: [CLX-261](https://instructure.atlassian.net/browse/CLX-261)

Add `renderIcons` prop to make the increase/decrease buttons customizable with externally defined icons
TEST PLAN:
Chromatic test for rendered icons and unit test to ensure unchanged behaviour

[CLX-261]: https://instructure.atlassian.net/browse/CLX-261?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/ui-number-input/src/NumberInput/props.ts [hunk 1]
```diff
@@ -226,7 +234,11 @@ const propTypes: PropValidators<PropKeys> = {
   onKeyDown: PropTypes.func,
   inputMode: PropTypes.oneOf(['numeric', 'decimal', 'tel']),
   textAlign: PropTypes.oneOf(['start', 'center']),
-  allowStringValue: PropTypes.bool
+  allowStringValue: PropTypes.bool,
+  renderIcons: PropTypes.shape({
```
Comment (HerrTopi) [inline packages/ui-number-input/src/NumberInput/props.ts:238]: Please add the "renderIcons" prop to the allowedProps list as well (see below in the code)

File: packages/ui-number-input/src/NumberInput/props.ts [hunk 2]
```diff
@@ -166,6 +166,14 @@ type NumberInputOwnProps = {
    * sets the input type to string and allows string as value
    */
   allowStringValue?: boolean
+
+  /**
+   * Definition of the icons to be rendered for increase and decrease buttons
```
Comment (HerrTopi) [inline packages/ui-number-input/src/NumberInput/props.ts:171]: Kinda nitpicky: Since this text acts as the documentation for the prop on the insUI docs page, I would rephrase it to something like this:

"Sets the icons to be rendered for increase and decrease buttons"

Comment (szalonna) [reply packages/ui-number-input/src/NumberInput/props.ts]: Fixed

**Review by ToMESSKa — APPROVED** [review]

**Review by matyasf — APPROVED** [review]
